lib/Parse/ParseType.cpp

```diff
@@ -633,12 +633,42 @@ ParserStatus Parser::parseGenericArguments(SmallVectorImpl<TypeRepr *> &Args,
   return makeParserSuccess();
 }
 
+/// Returns true if a base type for a qualified declaration name can be
+/// parsed.
+///
+/// Examples:
+///   'Foo.f' -> true
+///   'Foo.Bar.f' -> true
+///   'f' -> false, no base type
+bool Parser::canParseBaseTypeForQualifiedDeclName() {
```

Comment:
It would be great if  this can share some code with `canParseTypeIdentifier()`.
Also please move this function below line 1374.

Reply:
Thanks for the catch!

I moved `Parser::canParseBaseTypeForQualifiedDeclName` just below `Parser::canParseTypeIdentifier` to line 1562. I also removed the doc comment on the function definition to be consistent with the surrounding code.

Reply:
Code sharing done in https://github.com/apple/swift/pull/27545/commits/fa31c75aac4ac56c3b15e59b42bdc0c941cb2498.

I factored code into a new utility `Parser::canParseSimpleTypeIdentifier`, used by both `Parser::canParseTypeIdentifier` and `Parser::canParseBaseTypeForQualifiedDeclName`.